### DISCUSSION TITLE
Timouts in Flow Interview

### DISCUSSION THREAD

**Discussion by melissajhansen** [jamessimone/apex-rollup#275] [Q&A]
We've finally deployed Apex Rollups to my client's Production environment after quite a bit of testing, and the Production issue we're seeing is with timeouts that seem to occur during the flow interview.

Users are making bulk updates, and we see a CPU timeout in the flow itself, it doesn't make it far enough to show up in the APEX Rollup Logs.

The flow is the basic record-triggered flow that defers processing and uses the Perform Rollup__mdt-based rollup action.  We have 5 rollups configured for this object (Opportunity)

Has anyone else run into this?  I admittedly have much more experience with Apex than Flow, so perhaps I'm missing something basic about the flow setup?  The only thing I think I can do to address this would be to switch to an Apex triggered rollup so that we bypass flow entirely, but of course it would be a bummer to lose the declarative-configuration aspect.

If anyone has any insights, I'd appreciate it!

<img width="393" alt="image" src="https://user-images.githubusercontent.com/5881820/160652182-e14239ed-65fc-4fa4-bfe4-a3e7819854df.png">
<img width="739" alt="image" src="https://user-images.githubusercontent.com/5881820/160652539-ed914b71-94b1-41fd-9cae-79bb41b0b17e.png">

**Comment by jamessimone** [comment]
@melissajhansen I've pinged an old coworker of mine who is doing large bulk updates where the calling point for Apex Rollup is flow. We had made several improvements last year to the Flow engine part of Apex Rollup together to support their need for several thousand records to be updated at once. Which is to say - this is the first time I'm hearing of an issue like this in quite a while now, and I'm wondering if there are other (competing) flow automations which could be contributing to such a thing?

I'm awaiting word to see if there was anything notable about their setup that we can pass on to you (looking at your screenshots, everything looks great though ...). Otherwise, let me know what else I can do to assist in troubleshooting. There's nothing that immediately comes to mind, on my end, as far as how you could be hitting a limit exception so early on in the process - but while there may not be rollup logs getting generated, I would imagine that despite that, looking at a debug log might help to see what's going on under the hood.

**Reply by melissajhansen** [reply]
@jamessimone competing flows is definitely a potential culprit here.  They have other flows that are timing out intermittently and the overall complexity of their org is pretty high.  

I went ahead and switched them over to Apex-triggered rollups in Production for now, and I'm going to do some more trouble-shooting in their full sandbox.  Definitely let me know if you hear back on any special setup that I should be considering and I'll report back when I have a better idea if the rollup flow was a big offender CPU-wise, or if it was just the straw that broke the camel's back.

TX!

**Reply by jamessimone** [reply]
@melissajhansen I spoke with my friend yesterday and here's what they had to say:

>  From what I recall with the timeouts after, you worked to optimize things for us on the refresh. Our timeouts then stopped being rollup related, and became other apex/flow timeouts. If I recall, if it did make it to the roller flow, it ran fine. We did something in config too and the setting is escaping me. It was some size related option. [edit from James: this is the `Max Lookup Rows Before Batching` setting on Rollup Control] ... That definitely helped us. Biggest thing for me was intense debug log [perusing] to where we really nailed down that the rollup jobs never truly started and our issues were elsewhere. Once you made your last optimization for `refresh`, if it made it to the rollup action, it made it ... So my bet is, if rollup is timing out, that object has a ton of stuff getting kicked off [before] she does [...] the rollup.
